modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/RescheduleTask.java

```diff
@@ -100,18 +70,24 @@ private void doRun()
             ifPresent( duration -> {
                 try
                 {
-                    schedulerExecutorService.schedule( SchedulableTask.create().
+                    OsgiSupport.withService( SchedulerExecutorService.class, scheduler -> scheduler.schedule( SchedulableTaskImpl.create().
                         job( job ).
-                        build(), duration.isNegative() ? 0 : duration.toMillis(), TimeUnit.MILLISECONDS );
-                }
-                catch ( DuplicateTaskException e )
-                {
-                    LOG.debug( "[{}] job is scheduled already.", job.getName().getValue(), e );
+                        build(), duration.isNegative() ? 0 : duration.toMillis(), TimeUnit.MILLISECONDS ) );
                 }
                 catch ( Exception e )
                 {
                     LOG.warn( "[{}] job rescheduling failed.", job.getName().getValue(), e );
                 }
             } ) );
     }
+
+    private static Context adminContext()
+    {
+        return ContextBuilder.from( ContextAccessor.current() ).
+            authInfo( AuthenticationInfo.
+                copyOf( ContextAccessor.current().getAuthInfo() ).
```

Comment:
Why do you copy current auth info? What happens if current user is not su?

```
User.create().
                key( PrincipalKey.ofSuperUser() ).
                login( PrincipalKey.ofSuperUser().getId() ).
                build()
```

Reply:
Then it will be current user with ADMIN role, We use it only to fetch data, so there is no big difference who does it. But yes, RescheduleTask is a system task, let it be superUser.